internal/attestation/idkeydigest/idkeydigest.go

```diff
@@ -188,23 +190,26 @@ func (d *List) UnmarshalJSON(b []byte) error {
 		}
 		encodedDigests = []string{unmarshalledString}
 	}
-	if err := d.unmarshal(encodedDigests); err != nil {
+	res, err := UnmarshalHexString(encodedDigests)
+	if err != nil {
 		return fmt.Errorf("unmarshalling json: %w", err)
 	}
+	*d = *res
 	return nil
 }
 
-// unmarshal is a helper function for unmarshalling encodedIDKeyDigests into IDKeyDigests.
-func (d *List) unmarshal(encodedDigests encodedList) error {
+// UnmarshalHexString unmarshals a list of hex encoded ID key digest strings.
+func UnmarshalHexString(encodedDigests []string) (*List, error) {
+	res := &List{}
 	for _, encodedDigest := range encodedDigests {
 		if len(encodedDigest) != encodedDigestLength {
-			return fmt.Errorf("invalid digest length: %d", len(encodedDigest))
+			return nil, fmt.Errorf("invalid digest length: %d", len(encodedDigest))
 		}
 		digest, err := hex.DecodeString(encodedDigest)
 		if err != nil {
-			return fmt.Errorf("decoding digest: %w", err)
+			return nil, fmt.Errorf("decoding digest: %w", err)
 		}
-		*d = append(*d, digest)
+		*res = append(*res, digest)
 	}
-	return nil
+	return res, nil
 }
```

Comment:
```suggestion
func UnmarshalHexString(encodedDigests []string) (List, error) {
	res := make(List, 0, len(encodedDigests))
	for idx, encodedDigest := range encodedDigests {
		if len(encodedDigest) != encodedDigestLength {
			return nil, fmt.Errorf("invalid digest length: %d", len(encodedDigest))
		}
		digest, err := hex.DecodeString(encodedDigest)
		if err != nil {
			return nil, fmt.Errorf("decoding digest: %w", err)
		}
		res[idx] = digest
	}
	return res, nil
}
```
nit: this is a bit cleaner
Looking at the rest of the code, we also don't usually work on a pointer to `List`, but on it directly.